[feat] Drop support of pipeline stage method override (#1636)
eth-cscs/reframe: docs/migration_2_to_3.rst

ReFrame 2.20 introduced a new powerful mechanism for attaching arbitrary functions hooks at the different pipeline stages.
This mechanism provides an easy way to configure and extend the functionality of a test, eliminating essentially the need to override pipeline stages for this purpose.

> ReFrame 3.0 deprecates the old practice of overriding pipeline stage methods in favor of using pipeline hooks.
> From ReFrame 3.4, overriding pipeline stage methods is no longer allowed.

Reviewer: ```suggestion
ReFrame 3.0 deprecates the old practice of overriding pipeline stage methods in favor of using pipeline hooks and ReFrame 3.4 disables that by default.
```
